pycbc/events/stat.py

```diff
@@ -87,282 +93,229 @@ def single(self, trigs):
         numpy.ndarray
             The array of single detector values
         """
-        return ranking.get_newsnr(trigs)
-
-    def coinc(self, s0, s1, slide, step): # pylint:disable=unused-argument
-        """Calculate the coincident detection statistic.
-
-        Parameters
-        ----------
-        s0: numpy.ndarray
-            Single detector ranking statistic for the first detector.
-        s1: numpy.ndarray
-            Single detector ranking statistic for the second detector.
-        slide: (unused in this statistic)
-        step: (unused in this statistic)
+        return ranking.get_sngls_ranking_from_trigs(
+            trigs,
+            self.sngl_ranking,
+            **self.sngl_ranking_kwargs
+        )
 
-        Returns
-        -------
-        numpy.ndarray
-            Array of coincident ranking statistic values
+    def single(self, trigs): # pylint:disable=unused-argument
         """
-        return (s0 ** 2. + s1 ** 2.) ** 0.5
-
-    def coinc_lim_for_thresh(self, s0, thresh):
-        """Calculate the required single detector statistic to exceed
-        the threshold for each of the input triggers.
+        Calculate the necessary single detector information
 
         Parameters
         ----------
-        s0: numpy.ndarray
-            Single detector ranking statistic for the first detector.
-        thresh: float
-            The threshold on the coincident statistic.
+        trigs: dict of numpy.ndarrays, h5py group (or similar dict-like object)
+            Dictionary-like object holding single detector trigger information.
 
         Returns
         -------
         numpy.ndarray
-            Array of limits on the second detector single statistic to
-        exceed thresh.
+            The array of single detector values
         """
-        s1 = thresh ** 2. - s0 ** 2.
-        s1[s1 < 0] = 0
-        return s1 ** 0.5
-
-    def coinc_multiifo(self, s, slide, step, to_shift,
-                       **kwargs): # pylint:disable=unused-argument
-        """Calculate the coincident detection statistic.
-
-        Parameters
-        ----------
-        s: list
-            List of (ifo, single detector statistic) tuples
-        slide: (unused in this statistic)
-        step: (unused in this statistic)
-        to_shift: list
-            List of integers indicating what multiples of the time shift will
-        be applied (unused in this statistic)
+        err_msg = "This function is a stub that should be overridden by the "
+        err_msg += "sub-classes. You shouldn't be seeing this error!"
+        raise ValueError(err_msg)
```

Comment:
as noted in the main PR thread, this and analogous exceptions should preferably be `NotImplementedError`s.